### PR TITLE
Upgrade to TensorFlow 1.13.1

### PR DESCRIPTION
**Pull request by icywang86rui**
* Add 1.13.1 dockerfiles
* Lower process number for one of the horovod local mode test
    Too many processes running in parallel causes the test fail sometimes.

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docker/1.13.1/Dockerfile.cpu [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    curl -fSsL -O https://www.open-mpi.org/software/ompi/v3.1/downloads/openmpi-3.1.2.tar.gz && \
+    tar zxf openmpi-3.1.2.tar.gz && \
+    cd openmpi-3.1.2 && \
+    ./configure --enable-orterun-prefix-by-default && \
+    make -j $(nproc) all && \
+    make install && \
+    ldconfig && \
+    rm -rf /tmp/openmpi
+
+
```
Comment (laurenyu) [inline docker/1.13.1/Dockerfile.cpu]: don't need two empty newlines

File: docker/1.13.1/Dockerfile.cpu [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+ # Create a wrapper for OpenMPI to allow running as root by default
```
Comment (laurenyu) [inline docker/1.13.1/Dockerfile.cpu]: extra space before the pound sign

File: docker/1.13.1/Dockerfile.cpu [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        ln -s -f /usr/bin/$PYTHON_VERSION /usr/local/bin/python && \
+    rm -rf /var/lib/apt/lists/*
+
+ENV PYTHONDONTWRITEBYTECODE=1 PYTHONUNBUFFERED=1 PYTHONIOENCODING=UTF-8 LANG=C.UTF-8 LC_ALL=C.UTF-8
+
+RUN curl -fSsL -O https://bootstrap.pypa.io/get-pip.py && \
+    python get-pip.py --disable-pip-version-check --no-cache-dir "pip==18.1" && \
+    rm get-pip.py
+
+ARG framework_installable
```
Comment (laurenyu) [inline docker/1.13.1/Dockerfile.cpu]: can you group the args together?

File: docker/1.13.1/Dockerfile.cpu [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        tensorflow-1.13.1-py2.py3-none-any.whl \
+        horovod && \
+    pip install --no-cache-dir -U $framework_support_installable && \
+    rm -f tensorflow-1.13.1-py2.py3-none-any.whl && \
+    rm -f $framework_support_installable && \
+    pip uninstall -y --no-cache-dir \
+        markdown \
+        tensorboard
+
+ENV SAGEMAKER_TRAINING_MODULE sagemaker_tensorflow_container.training:main
```
Comment (laurenyu) [inline docker/1.13.1/Dockerfile.cpu]: add a newline to the end of the file

File: setup.py [hunk 5]
```diff
@@ -53,7 +53,7 @@ def read(fname):
                       'pandas', 'Pillow', 'h5py'],
     extras_require={
         'test': ['tox', 'flake8', 'pytest', 'pytest-cov', 'pytest-xdist', 'mock',
-                 'sagemaker>=1.15.2', 'tensorflow', 'docker-compose'],
+                 'sagemaker>=1.19.1', 'tensorflow', 'docker-compose'],
```
Comment (laurenyu) [inline setup.py]: let's pin this to reduce the blast radius of potential issues with Python SDK releases

File: docker/1.13.1/Dockerfile.cpu [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+RUN curl -fSsL -O https://bootstrap.pypa.io/get-pip.py && \
+    python get-pip.py --disable-pip-version-check --no-cache-dir "pip==18.1" && \
+    rm get-pip.py
+
+COPY $framework_installable tensorflow-1.13.1-py2.py3-none-any.whl
+COPY $framework_support_installable .
+
+RUN pip install --no-cache-dir -U \
+        keras==2.2.4 \
+        mpi4py==3.0.1 \
```
Comment (gautamkmr) [inline docker/1.13.1/Dockerfile.cpu]: why do you need this when you have openmpi ?

File: docker/1.13.1/Dockerfile.gpu [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+    rm get-pip.py
+
+WORKDIR /
+
+COPY $framework_installable tensorflow-1.13.1-py2.py3-none-any.whl
+COPY $framework_support_installable .
+
+RUN pip install --no-cache-dir -U \
+    keras==2.2.4 \
+    mpi4py==3.0.1 \
```
Comment (gautamkmr) [inline docker/1.13.1/Dockerfile.gpu]: same comment as above

File: docker/1.13.1/Dockerfile.cpu [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+    openssh-server \
+    ca-certificates \
+    curl \
+    && add-apt-repository ppa:deadsnakes/ppa -y \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install Open MPI
+RUN mkdir /tmp/openmpi && \
+    cd /tmp/openmpi && \
+    curl -fSsL -O https://www.open-mpi.org/software/ompi/v3.1/downloads/openmpi-3.1.2.tar.gz && \
```
Comment (mirocody) [inline docker/1.13.1/Dockerfile.cpu]: Can we install openmpi v4.0?

File: docker/1.13.1/Dockerfile.gpu [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+        cuda-cublas-dev-10-0 \
+        cuda-cudart-dev-10-0 \
+        cuda-cufft-dev-10-0 \
+        cuda-curand-dev-10-0 \
+        cuda-cusolver-dev-10-0 \
+        cuda-cusparse-dev-10-0 \
+        curl \
+        libcudnn7=7.4.1.5-1+cuda10.0 \
+        # TensorFlow doesn't require libnccl anymore but Open MPI still depends on it
+        libnccl2 \
```
Comment (mirocody) [inline docker/1.13.1/Dockerfile.gpu]: As this is the dependencies of OpenMPI, and openmpi is pinned with version, could you please pin this version as well?

File: docker/1.13.1/Dockerfile.gpu [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+        libnccl-dev \
```
Comment (mirocody) [inline docker/1.13.1/Dockerfile.gpu]: same above.

**Comment by andremoeller** [comment]
Can you add a `_wait_until_master_is_up` function that's called before `_wait_until_master_is_down`?

This is so that non-master nodes don't immediately exit if they come up before master starts. This is a bigger risk with jobs with many instances.

**Pull request merged**